zephyr/esp_shared/include/stubs.h

```diff
@@ -41,6 +41,10 @@
 #define CONFIG_ESP_CONSOLE_UART_NUM 2
 #define CONFIG_ESP_CONSOLE_UART_BAUDRATE DT_PROP(DT_NODELABEL(uart2), current_speed)
 
+#else
+
+#define CONFIG_ESP_CONSOLE_UART_NUM -1
```

Comment:
This is missing and required. When uart is disabled, i.e., using USB interface, build will fail. This fixes it.